app/models/spree/gateway/amazon.rb

```diff
@@ -183,6 +183,16 @@ def void(response_code, gateway_options)
       return ActiveMerchant::Billing::Response.new(true, "Success", response)
     end
 
+    def cancel(response_code)
+      payment = Spree::Payment.find_by(response_code: response_code)
+      order = payment.order
+      load_amazon_mws(payment.source.order_reference)
+
+      response = @mws.refund(response_code, order.number, payment.credit_allowed, payment.currency)
```

Comment:
I think we actually want to do the same thing here as we're doing in `void` -- either cancel or refund.  That's what [solidusio/solidus_braintree](https://github.com/solidusio/solidus_braintree/blob/81176dcdab70390870e2111be8ec1933b99782a9/app/models/solidus/gateway/braintree_gateway.rb#L169-L177) does at least.  It might actually be that we're doing an odd thing with `void` (perhaps `void` should only cancel and not refund).

But in any case for now maybe we could extract that common code for both of these methods and do the same thing in both places?